models/models.go

```diff
@@ -156,86 +146,109 @@ type StepRunResultsModel struct {
 	ExitCode   int                         `json:"exit_code" yaml:"exit_code"`
 }
 
-// TestResultStepInfo ...
-type TestResultStepInfo struct {
-	ID      string `json:"id" yaml:"id"`
-	Version string `json:"version" yaml:"version"`
-	Title   string `json:"title" yaml:"title"`
-	Number  int    `json:"number" yaml:"number"`
+// StepError ...
+type StepError struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
 }
 
-// StepRunStatus ...
-type StepRunStatus int
-
-// Reason ...
-func (s StepRunStatus) Reason(exitCode int) string {
-	switch s {
-	case StepRunStatusCodeSuccess, StepRunStatusCodeSkipped, StepRunStatusCodeSkippedWithRunIf:
-		return ""
-	case StepRunStatusCodeFailed, StepRunStatusCodePreparationFailed, StepRunStatusCodeFailedSkippable:
-		return fmt.Sprintf("exit code: %d", exitCode)
+func (s StepRunResultsModel) StatusReasons() (string, []StepError) {
```

Comment:
The method returns more than what the name suggest. I do not have a great suggestion but I would rename it to `StatusReasonAndErrors` or `FinalStatus` or something else.

Reply:
We decided to use this name because both the status_reason and the errors are reasonings for the given status.